src/WelcomeSmart.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import { Form, Button } from 'semantic-ui-react';
+import { SMARTWrap } from './FHIRClientWrapper';
+import Loading from './Loading';
+
+const redirectURI = window.location.href + 'standalone'
+
+class WelcomeSmart extends Component {
+
+  constructor(props) {
+    super(props);
+    // See if we have state from previous use stored in local browser storage
+    const localState = localStorage['stateWelcomeSmart'];
+    this.state = localState ? JSON.parse(localState) : { fhirServer: '', clientId: '', secret: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const target = event.target;
+    this.setState({ [target.name]: target.value }, () => {
+      // Store all state in local browser storage too to preserve between uses
+      localStorage['stateWelcomeSmart'] = JSON.stringify(this.state);
```

Comment:
Nit-picky: Any reason for serializing/deserializing this using `JSON.stringify`/`JSON.parse` and storing in a single key, instead of using key/values for each of `fhirServer`, `clientId`, and `secret` (`localStorage.setItem(key, value)`, `localStorage.getItem(key)`)?

Reply:
I think this way is a bit less code and also transparently scales to additional fields or supports changes in fields.

Reply:
Fair enough, just need to remember to add default values for any additional fields above in line 15, when doing the `JSON.parse` on whatever is in `localStorage['stateWelcomeSmart']`.

Reply:
Good call, that would lead to a pretty nasty bug that would only show up for people who had something in local storage... maybe something like

```
    // See if we have state from previous use stored in local browser storage
    const defaultState = { fhirServer: '', clientId: '', secret: '' };
    const localState = localStorage['stateWelcomeSmart'];
    if (localState) {
      this.state = Object.assign({}, defaultState, localState);
    } else {
      this.state = defaultState;
    }
```